Sources/StreamChatUI/ChatChannelList/ChatChannelListItemView.swift

```diff
@@ -188,6 +198,30 @@ open class ChatChannelListItemView: _View, ThemeProvider, SwiftUIRepresentable {
 
         unreadCountView.content = content?.channel.unreadCount ?? .noUnread
         unreadCountView.invalidateIntrinsicContentSize()
+        
+        let checkmarkContent: ChatMessageDeliveryStatusCheckmarkView.Content? = content.flatMap {
+            guard
+                let previewMessage = $0.channel.previewMessage,
+                let deliveryStatus = previewMessage.deliveryStatus
+            else { return nil }
+            
+            let canShow = components.messageLayoutOptionsResolver.canShowDeliveryStatus(
+                for: previewMessage,
+                in: $0.channel,
+                allowWhenFailed: true
+            )
```

Comment:
IMO the resolver shouldn't be used in the `ChatChannelListItemView`. I would prefer to separate it into 2 functions and remove the `allowWhenFailed`. The reason is that the `allowWhenFailed` looks like a code smell. In reality, the logic for showing the delivery status is different in both places (Although they diverge in just one detail in this case). 

There are multiple ways we can fix this:
- Keep the `canShowDeliveryStatus()` in the Resolver and create a new one in `ChatChannelListItemView`.
- Create a protocol `DeliveryStatusShowing` and create a `MessagePreviewDeliveryStatus` and a `MessageDeliveryStatus`

The second approach might be an overkill for this case, but what I mean is that I think we need to distinguish 2 delivery status logics, one from a regular message, and one from a preview message. I believe that adding a flag in the argument of that function it won't scale very well 🤔

Reply:
Makes sense 👍

Reply:
Done ✅